client/components/new-expense-detail.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import ToggleSwitch from './toggle-switch';
+
+export default class NewExpenseDetail extends React.Component {
+  constructor(props) {
+    super(props);
+    this.makeOptions = this.makeOptions.bind(this);
+  }
+
+  makeOptions(array) {
+    if (!array) {
```

Comment:
This conditional is confusing. Arrays are always truthy, even if they are empty.

If somehow this array is going to be something other than an array, I recommend adjusting your code so that this value is _always_ an array. This is a rare instance of "defensive programming" that I recommend. Try to pass consistently shaped data if you can. Particularly when it comes to arrays. Either it has `0` length or it doesn't.